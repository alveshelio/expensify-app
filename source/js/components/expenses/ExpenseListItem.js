import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import moment from 'moment';

const ExpenseListItem = ({
  id,
  description,
  amount,
  note,
  createdAt,
  onRemoveHandle,
}) => {
  return (
    <div className='expense-list-item'>
      <span><Link to={`/edit/${id}`}>{description}</Link></span>
      {note && <span>Note: {note}</span>}
      <span>Amount: {amount}</span>
      {createdAt && <span>Created On: {moment(createdAt).format('MMM Do, YYYY')}</span>}
      <button onClick={() => onRemoveHandle(id)}>Remove</button>
      <hr />
    </div>
  );
};

ExpenseListItem.propTypes = {
  id: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  amount: PropTypes.number.isRequired,
  note: PropTypes.string,
  createdAt: PropTypes.number.isRequired,
  onRemoveHandle: PropTypes.func.isRequired,
};


export default ExpenseListItem;
