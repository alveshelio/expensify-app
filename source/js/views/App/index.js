import React from 'react';

const App = () => (
  <div>
    <h1>App is running on port 3000</h1>
  </div>
);

export default App;
