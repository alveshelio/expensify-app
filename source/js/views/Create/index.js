import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import ExpenseForm from '../../components/expenses/forms/ExpenseForm';
import { addExpense } from '../../actions/expensesActions';

export class CreateExpensePage extends Component {
  onSubmit = (expense) => {
    this.props.onSubmit(expense);
    this.props.history.push('/');
  };
  render() {
    return (
      <div>
        <h1>Create</h1>
        <ExpenseForm
          onSubmit={this.onSubmit}
        />
      </div>
    );
  }
}

CreateExpensePage.propTypes = {
  onSubmit: PropTypes.func.isRequired,
  history: PropTypes.shape({
    push: PropTypes.func.isRequired,
  }).isRequired,
};

const mapDispatchToProps = dispatch => ({
  onSubmit: expense => dispatch(addExpense(expense)),
});

export default connect(undefined, mapDispatchToProps)(CreateExpensePage);
