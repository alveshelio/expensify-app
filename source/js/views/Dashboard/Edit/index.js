import React from 'react';

const Edit = () => (
  <div>
    <h1>Edit</h1>
    <p>Dashboard Edit Expense</p>
  </div>
);

export default Edit;
