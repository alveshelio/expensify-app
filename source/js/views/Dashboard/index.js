import React from 'react';
import DashboardRoutes from '../../routes/DashboardRoutes';

const Dashboard = () => (
  <div>
    <h1>Dashboard</h1>
    <DashboardRoutes />
  </div>
);

export default Dashboard;
