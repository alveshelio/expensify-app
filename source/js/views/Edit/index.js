import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import ExpenseForm from '../../components/expenses/forms/ExpenseForm';
import { editExpense } from '../../actions/expensesActions';

export class EditExpensePage extends Component {
  onSubmit = expenseToEdit => {
    const expenseId = this.props.match.params.id;
    // expenseToEdit is being passed in the ExpenseForm when we
    // call onSubmit and pass the object from state
    this.props.editExpense(expenseId, expenseToEdit);
    this.props.history.push('/');
  };

  render() {
    const { expense } = this.props;

    return (
      <div>
        <h2>Edit expense {expense.description}</h2>
        <ExpenseForm
          onSubmit={this.onSubmit}
          expense={expense}
        />
      </div>
    );
  }
}

EditExpensePage.propTypes = {
  match: PropTypes.shape({
    params: PropTypes.shape({
      id: PropTypes.string.isRequired,
    }).isRequired,
  }).isRequired,
  editExpense: PropTypes.func.isRequired,
  expense: PropTypes.shape({
    id: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    amount: PropTypes.number.isRequired,
    note: PropTypes.string,
    createdAt: PropTypes.number,
  }).isRequired,
  history: PropTypes.shape({
    push: PropTypes.func.isRequired,
  }).isRequired,
};

const mapsStateToProps = (state, { match }) => ({
  expense: state.expenses.find(expense => expense.id === match.params.id),
});

const mapDispatchToProps = dispatch => ({
  editExpense: (expenseId, expenseToEdit) => dispatch(editExpense(expenseId, expenseToEdit)),
});

export default connect(mapsStateToProps, mapDispatchToProps)(EditExpensePage);
