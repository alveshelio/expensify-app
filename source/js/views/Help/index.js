import React from 'react';

const Help = (props) => {
  console.log('props', props);
  return (
    <div>
      <h1>Help Page</h1>
    </div>
  );
};

export default Help;
