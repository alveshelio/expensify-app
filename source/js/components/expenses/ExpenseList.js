import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import ExpenseListItem from './ExpenseListItem';
import selectExpenses from '../../selectors/expensesSelectors';
import { removeExpense } from '../../actions/expensesActions';

export const ExpenseList = ({ expenses, filters, dispatch }) => {
  return (
    <div>
      <h1>ExpenseList</h1>
      {
        expenses.length > 0 ?
          expenses.map(expense =>
            (<ExpenseListItem
              key={expense.id}
              {...filters}
              {...expense}
              onRemoveHandle={id => dispatch(removeExpense(id))}
            />)) :
          <p>You don&apos;t have any expenses yet</p>
      }
    </div>
  );
};

ExpenseList.propTypes = {
  expenses: PropTypes.arrayOf(PropTypes.shape({
    description: PropTypes.string.isRequired,
    amount: PropTypes.number.isRequired,
    note: PropTypes.string,
    createdAt: PropTypes.number.isRequired,
  }).isRequired).isRequired,
  filters: PropTypes.shape({
    text: PropTypes.string,
    sortBy: PropTypes.string,
    startDate: PropTypes.number,
    endDate: PropTypes.number,
  }),
  dispatch: PropTypes.func,
};

const mapStateToProps = state => ({
  expenses: selectExpenses(state.expenses, state.filters),
});

export default connect(mapStateToProps)(ExpenseList);
