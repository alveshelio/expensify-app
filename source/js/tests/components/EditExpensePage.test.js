/* eslint no-undef: "off" */

import React from 'react';
import { shallow } from 'enzyme';
import { EditExpensePage } from '../../views/Edit';

import expenses from '../../fixtures/expenses';

let onSubmit;
let history;
let wrapper;
let match;

beforeEach(() => {
  onSubmit = jest.fn();
  history = { push: jest.fn() };
  match = {
    params: {
      id: expenses[0].id,
    },
  };
  wrapper = shallow(<EditExpensePage
    editExpense={onSubmit}
    history={history}
    expense={expenses[0]}
    match={match}
  />);
});

test('should render Edit Expense Page correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should handle onSubmit', () => {
  wrapper.find('ExpenseForm').prop('onSubmit')(expenses[0]);
  expect(history.push).toHaveBeenLastCalledWith('/');
  expect(onSubmit).toHaveBeenLastCalledWith(expenses[0].id, expenses[0]);
});
