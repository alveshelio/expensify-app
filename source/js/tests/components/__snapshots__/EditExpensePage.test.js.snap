// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Edit Expense Page correctly 1`] = `
<div>
  <h2>
    Edit expense 
    Watter Bill
  </h2>
  <ExpenseForm
    expense={
      Object {
        "amount": 4554,
        "createdAt": -1209600000,
        "description": "Watter Bill",
        "id": "1",
        "note": "The watter is expensive",
      }
    }
    onSubmit={[Function]}
  />
</div>
`;
